Vector.c

```diff
@@ -88,13 +104,16 @@ int Vector_size(const Vector* this) {
 void Vector_prune(Vector* this) {
    assert(Vector_isConsistent(this));
    if (this->owner) {
-      for (int i = 0; i < this->items; i++)
+      for (int i = 0; i < this->items; i++) {
          if (this->array[i]) {
             Object_delete(this->array[i]);
-            //this->array[i] = NULL;
+            this->array[i] = NULL;
```

Comment:
NB: Not strictly thread-safe. But since we are single-threaded this is not a problem.
Better approach would be local variable to hold the item, remove it from the list and then `Object_delete''ing it.

Reply:
Yeah, nothing here is thread-safe and using a temporary variable won't fix that. Also, the compiler would have to re-order these statements for this to be invalid, which would be a bug.

Reply:
The compiler is allowed to write the `NULL` into the memory before it actully performs the call to `Object_delete()` IF it keeps the old contents of that pointer around as the argument to that call.